test/framework/src/main/java/org/elasticsearch/cluster/node/TestDiscoveryNode.java

```diff
@@ -9,47 +9,86 @@
 package org.elasticsearch.cluster.node;
 
 import org.elasticsearch.Version;
+import org.elasticsearch.common.UUIDs;
 import org.elasticsearch.common.transport.TransportAddress;
-import org.elasticsearch.test.ESTestCase;
 
+import java.security.AccessControlException;
 import java.util.Collections;
 import java.util.Map;
 import java.util.Set;
+import java.util.UUID;
 
+import static org.apache.lucene.tests.util.LuceneTestCase.random;
+import static org.elasticsearch.test.ESTestCase.buildNewFakeTransportAddress;
+
+/**
+ * Various shortcuts to constructors in {@link DiscoveryNode}.
+ */
 public class TestDiscoveryNode {
 
+    private static String newEphemeralId() {
+        try {
+            return UUIDs.randomBase64UUID(random());
+        } catch (AccessControlException e) {
+            // don't have SM permissions to access thread group context - probably part of a static init
+            return UUID.randomUUID().toString();
+        }
+    }
+
     public static DiscoveryNode create(String id) {
+        return create(id, buildNewFakeTransportAddress());
```

Comment:
There is quite an explosion of creation methods here, adding to an already large number of create methods. This seems to be moving complexity of the X number of ctors in DiscoveryNode to the this test class, and in the process amplifying it.

Could we consider a TestDiscoverNodeBuilder for the more complicated cases? The most common cases can have direct create methods (maybe one or two?) here, but the rest could use a builder. The difference between the production builder and this one is that the defaults would be test driven, ie fake transport addresses by default, etc.

Reply:
Yeah, I have been considering that. Now that fake address is included in this class it does tip it over the edge.

Seems like a good thing for a separate PR after this one, to avoid complicating this one too much?

Reply:
Wouldn't it eliminate the need for a lot of the create methods added here?

Reply:
Yes, but there's several other related changes already in here, so the combined one will be large. I've started on it in #95978

Reply:
Possibly, although I would try to decouple *adding* the builder from actually utilizing it, in order to decrease the surface area of each PR (making each more reviewable).

Reply:
Adding the builder & utilising it are separate commits, so I can slice & dice it however is needed.

Reply:
This PR is quite large (150 files). If most of these modifications will just change to the builder, it would seem a better use of time to get the builder in and then adjust this PR as a followup, IMO.

Reply:
https://github.com/elastic/elasticsearch/pull/96062